testctrl/svc/orch/reservation.go

```diff
@@ -132,16 +125,16 @@ func (rm *ReservationManager) Reserve(session *types.Session) error {
 	return nil
 }
 
-// Return accepts a session that has reserved machines that are no longer needed. It increases the
-// number of available machines to allow other sessions to provision resources. It does not actually
-// return, tear down or delete any machine or resources.
+
+// Unreserve increases the number of available machines by the number of machines a session
+// required. Essentially, it reverses the actions of the Reserve function.
 //
-// This method makes an assumption that Reserve has been called on the session, performing no checks
-// that the available machines in each pool do not exheed their capacities.
+// This method does not ensure that Reserve has been called on the session. If the caller
```

Comment:
Perhaps the method should track previous calls. Perhaps that should not be done per session but by returning a reservation id when making a reservation, and then requiring that reservation id to unreserve. This would make this interface more like a real reservation mechanism. That being said, this could be done in a later check-in as an enhancement.

Reply:
will address in later PR after some other things

Reply:
If the process dies, there is no recovery even with the id. This is more like a "rough bookkeeper" than a fully-fledged reservation system. If it were a full reservation system, it would continually communicate with kubernetes. I believe the constant communication would be overkill for a prototype and delay a version for infra guys.

Reply:
This comment mainly serves as a warning to not throw it in a defer block or call it after Reserve returns an error.

Reply:
Let's handle this in a future PR.